src/hvqm.c

```diff
@@ -199,68 +193,96 @@ static int process_info(struct hle_t* hle, uint8_t* base, int16_t* out)
     return 1;
 }
 
-#define SATURATE(x) ((unsigned int) x <= 31 ? x : (x < 0 ? 0 : 31))
-static uint16_t YCbCr_to_BGRA5551(int16_t Y, int16_t Cb, int16_t Cr, uint8_t alpha)
+#define SATURATE8(x) ((unsigned int) x <= 255 ? x : (x < 0 ? 0: 255))
+static struct RGBA YCbCr_to_RGBA(int16_t Y, int16_t Cb, int16_t Cr, uint8_t alpha)
```

Comment:
Not a review per see, but just a note: This function is always called from a 4 sized loop (itself inside a loop of a loop), I wonder if it wouldn't be more efficient to check `arg.chroma_step_v` and allocate fixed size array before the loops so you can make this function deals with array and do the component maths at once.

If have no idea about performance anyway so just a note.

Reply:
Well I organized loops to actually have this function to only deal with one pixel at a time :-)
I think it's easier to understand that way but it's just my opinion ofc.

Reply:
Yes, it's definitely easier to understand, my point was about performance, but I have no clue.